### ISSUE TITLE
sequencer: attach next valid nonce to INVALID_NONCE tx response

### ISSUE THREAD

**Issue by itamarreif** [astriaorg/astria#529]
This would make it easier (e.g. for the Composer) to resubmit transaction that fail submission due to invalid nonce.

Currently (after #459 is merged) an `INVALID_NONCE` `AbciCode` response triggers a nonce query and then a new submission under the new nonce. This would obviate the need for the nonce query and would allow the Composer to use the nonce returned in the `INVALID_NONCE` response.

**Comment by itamarreif** [comment]
Closing as this is no longer relevant after adding the app-side mempool logic
